src/main/java/gregtech/common/covers/CoverPump.java

```diff
@@ -66,14 +68,24 @@ protected void setTransferRate(int transferRate) {
     }
 
     protected void adjustTransferRate(int amount) {
-        setTransferRate(MathHelper.clamp(transferRate + amount, 1, maxFluidTransferRate));
+        if (this.bucketMode == BucketMode.BUCKET)
+            setTransferRate(MathHelper.clamp(transferRate + amount * 1000, 1, maxFluidTransferRate));
+        else if (this.bucketMode == BucketMode.MILLI_BUCKET)
+            setTransferRate(MathHelper.clamp(transferRate + amount, 1, maxFluidTransferRate));
```

Comment:
I think we should use this one as it is clear enough but does not have that much boiler plate on it
```suggestion
        amount *= this.bucketMode == BucketMode.BUCKET ? 1000 : 1;
        setTransferRate(MathHelper.clamp(transferRate + amount, 1, maxFluidTransferRate));
```